service-proxy.js

```diff
@@ -979,6 +1047,80 @@ function setReapPeersPeriod(period) {
     self.peerReaper.setInterval(period);
 };
 
+ServiceDispatchHandler.prototype.setPrunePeersPeriod =
+function setPrunePeersPeriod(period) {
+    // period === 0 means never / disabled, and is the default
+    var self = this;
+
+    self.peerPruner.setInterval(period);
+};
+
+ServiceDispatchHandler.prototype.pruneSinglePeer =
+function pruneSinglePeer(hostPort, lastRefresh) {
+    var self = this;
+
+    var peer = self.channel.peers.get(hostPort);
+    if (!peer) {
+        return;
+    }
+
+    if (peer.draining) {
+        self.logger.info('skipping peer prune drain, already draining', self.extendLogInfo({
+            peer: peer.hostPort,
+            priorDrainReason: peer.drainReason
+        }));
+        return;
+    }
+
+    // TODO: stat?
+    self.logger.info('draining ex-affinity peer', self.extendLogInfo({
```

Comment:
need to fix this log message, now that I've separated this feature out of ex-affinity